src/runtime/uv/tcp.h

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright (c) 2025 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Sofia Rodrigues
+*/
+#pragma once
+#include <lean/lean.h>
+#include "runtime/uv/event_loop.h"
+#include "runtime/uv/net_addr.h"
+#include "runtime/object_ref.h"
+
+namespace lean {
+
+static lean_external_class* g_uv_tcp_socket_external_class = NULL;
+void initialize_libuv_tcp_socket();
+
+#ifndef LEAN_EMSCRIPTEN
+#include <uv.h>
+
+// Structure for managing a single TCP socket object, including promise handling,
+// connection state, and read/write buffers.
+typedef struct {
+    uv_tcp_t*      m_uv_tcp;           // LibUV TCP handle.
+    lean_object*   m_promise_accept;   // The associated promise for asynchronous results for accepting new sockets.
+    lean_object*   m_promise_read;     // The associated promise for asynchronous results for reading from the socket.
+    lean_object*   m_promise_shutdown; // The associated promise for asynchronous results to shutdown the socket.
+    lean_object*   m_client;           // Cached client that is going to be used in the next accept.
+    lean_object*   m_byte_array;       //  Buffer for storing data received via `recv_start`.
+} lean_uv_tcp_socket_object;
+
+// =======================================
```

Comment:
This header promises some functions that don't exist anymore and the prototypes aren't perfectly in sync with the state in tcp.cpp.